wordpress-mobile/WordPress-Android
# Add new Subscribers tab

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/20680


This PR adds in some basic groundwork for the new Subscribers tab, which is feature flagged behind `stats_traffic_subscribers_tab`. It is currently using `mapInsights` from `UiModelMapper.kt` as a placeholder for now, which is why you will see the screen below: 

<img width="286" alt="Screenshot 2024-04-22 at 4 43 22 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/809614f5-9d2d-4bfe-b338-405515d2c54b">

I've left the uses cases empty for now as that work can be done in a separate PR. 

-----

## To Test:
1. With the `stats_traffic_subscribers_tab` feature flag on, ensure you see the `Subscribers` tab as it appears in the screenshot above. 
2. Accessing the `Subscribers` tab should fire the `stats_subscribers_accessed` tracked event.
3. With the `stats_traffic_subscribers_tab` feature flag off, ensure you don't see the `Subscribers` tab and the `weeks/days/months` tabs appear as expected.



-----

## Regression Notes

1. Potential unintended areas of impact

    - States related to the `stats_traffic_subscribers_tab` feature flag

4. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

5. What automated tests I added (or what prevented me from doing so)

    - Tests will be updated in future PRs as development continues

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
